pom.xml

```diff
@@ -780,15 +780,6 @@
             <includeTestResources>false</includeTestResources>
             <includeTestSourceDirectory>true</includeTestSourceDirectory>
           </configuration>
-          <executions>
-            <execution>
-              <id>checkstyle</id>
-              <phase>validate</phase>
-              <goals>
-                <goal>check</goal>
-              </goals>
-            </execution>
-          </executions>
```

Comment:
I'm trying to understand Maven a bit better, and am puzzled by this: could you explain how you came to the conclusion that it was not necessary?

The "Checking for Violations as Part of the Build" section of the [usage page](https://maven.apache.org/plugins/maven-checkstyle-plugin/usage.html) includes that stanza.  Did you have to dig into the plugin's source code?

(I understand that the other dependency/configuration removals correspond to global defaults set in the `pluginManagement` section, but not this `executions` business.)

Reply:
This was removed because it literally did nothing. It defined an execution inside the `build/pluginManagement/plugins` section, and was never activated in a `build/plugins` section. The `<pluginManagement />` element is a way of early binding configuration to a plugin that is specified later. But, since the only execution of this plugin is on the command-line and not in a `build/plugins` section elsewhere in the `pom.xml` file, this literally did nothing.

The reason I removed it as part of this PR, rather than as a separate issue, is because the execution of checkstyle is part of the CI workflow for ZooKeeper, which was the focus of this PR.

Normally, you would specify it similarly to:

```xml
<build>
  <pluginManagement>
    <plugins>
      <plugin>
        <groupId>my.group</groupId>
        <artifactId>my.plugin</artifactId>
        <version />
        <!-- configuration for all executions of this plugin -->
        <configuration />
      </plugin>
    </plugins>
  </pluginManagement>
  <plugins>
    <plugin>
      <groupId>my.group</groupId>
      <artifactId>my.plugin</artifactId>
      <executions>
        <execution>
          <id />
          <goals />
          <!-- execution-specific configuration -->
          <configuration />
        </execution>
      </executions>
    </plugin>
  </plugins>
</build>
```

But we had:

```xml
<build>
  <pluginManagement>
    <plugins>
      <plugin>
        <groupId>my.group</groupId>
        <artifactId>my.plugin</artifactId>
        <version />
        <executions />
      </plugin>
    </plugins>
  </pluginManagement>
  <!-- this section was missing, so the execution in the above was never activated
  <plugins>
    <plugin>
      <groupId>my.group</groupId>
      <artifactId>my.plugin</artifactId>
    </plugin>
  </plugins>
  -->
</build>
```

I hope that helps explain it.

Reply:
I'm happy to answer more Maven stuff... I've learned quite a bit about it and love sharing. Ask any time.

Reply:
@ctubbsii: Very clear explanation; thank you! And the zero remaining "executions" seem to be enough, indeed, if we continue triggering that plugin from the command-line.

Reply:
If ZK ever decides to bake it in to the build lifecycle instead of executing it on the command-line, I can either help perform that work or review it. :smiley_cat: